### PR TITLE
Update BERT to take advantage of bias param in MultiHeadAttention

### PR DESCRIPTION
**Pull request by jbarrow**
Reduced duplication in the implementation. Tested on my machine (with a local mlx build) and gives the same results as before and as the HF implementation.

### CODE REVIEW

File: bert/model.py [hunk 1]
```diff
@@ -186,11 +117,26 @@ def __call__(
...
+
         y = self.encoder(x, attention_mask)
         return y, mx.tanh(self.pooler(y[:, 0]))
+    
+
+    def convert_mask_to_additive_causal_mask(
+        self, mask: mx.array, dtype: mx.Dtype = mx.float32
+    ) -> mx.array:
+        mask = mask == 0
+        mask = mask.astype(dtype) * -1e9
```
Comment (awni) [inline bert/model.py:137]: BTW if the mask is 0/s and 1/s you could do `mx.log(mask).astype(dtype)` instead.

**Review by awni — APPROVED** [review]
Love to see all those red lines. I'll wait till we do the next release then merge this.

**Comment by jbarrow** [comment]
A few more red lines thanks to your mask suggestion. 😄

Before you merge, but after there's a new MLX release I'll set a minimum `mlx` version in `requirements.txt`.

**Comment by jbarrow** [comment]
Updated `requirements.txt`!
